### PR TITLE
Get cluster

### PR DESCRIPTION
**Pull request by carsonmh**
Feature #183 

Makes function that returns Cluster object based on the cluster name and namespace
Tested using unit tests

### CODE REVIEW

File: src/codeflare_sdk/cluster/cluster.py [hunk 1]
```diff
@@ -337,6 +337,67 @@ def get_current_namespace():  # pragma: no cover
         return "default"
 
 
+def get_cluster(cluster_name: str, namespace: str = "default"):
+    try:
+        config.load_kube_config()
+        api_instance = client.CustomObjectsApi()
+        rcs = api_instance.list_namespaced_custom_object(
```
Comment (KPostOffice) [inline src/codeflare_sdk/cluster/cluster.py:344]: https://github.com/kubernetes-client/python/blob/045b189a282f5165426b841e9e3b16b2739d3837/kubernetes/client/api/custom_objects_api.py#L1580

Can you use this directly?

File: src/codeflare_sdk/cluster/cluster.py [hunk 2]
```diff
@@ -337,6 +337,67 @@ def get_current_namespace():  # pragma: no cover
...
+                    0
+                ]["resources"]["limits"]["nvidia.com/gpu"],
+                instascale=True if machine_types else False,
+                image=rc["spec"]["workerGroupSpecs"][0]["template"]["spec"][
+                    "containers"
+                ][0]["image"],
+                local_interactive=local_interactive,
+            )
+            return Cluster(cluster_config)
+    raise FileNotFoundError(
```
Comment (KPostOffice) [inline src/codeflare_sdk/cluster/cluster.py:396]: If you are able to use the function suggested above then the error handling can be removed here.

File: src/codeflare_sdk/cluster/cluster.py [hunk 3]
```diff
@@ -337,6 +337,67 @@ def get_current_namespace():  # pragma: no cover
...
+                gpu=rc["spec"]["workerGroupSpecs"][0]["template"]["spec"]["containers"][
+                    0
+                ]["resources"]["limits"]["nvidia.com/gpu"],
+                instascale=True if machine_types else False,
+                image=rc["spec"]["workerGroupSpecs"][0]["template"]["spec"][
+                    "containers"
+                ][0]["image"],
+                local_interactive=local_interactive,
+            )
+            return Cluster(cluster_config)
```
Comment (KPostOffice) [inline src/codeflare_sdk/cluster/cluster.py:395]: This should be broken out into it's own function in order to improve readability.

Something like:
```python
def k8_cluster_obj_2_sdk_cluster_obj(rc):
```

**Review by KPostOffice — COMMENTED** [review]
Looks good. Just a couple of comments.

File: src/codeflare_sdk/cluster/cluster.py (hunk 3 above)
Comment (KPostOffice) [reply src/codeflare_sdk/cluster/cluster.py:395]: maybe even a function in `Cluster` like `from_k8_cluster_object(rc)`

File: src/codeflare_sdk/cluster/cluster.py (hunk 1 above)
Comment (carsonmh) [reply src/codeflare_sdk/cluster/cluster.py:344]: I get `urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=80): Max retries exceeded with url: /apis/ray.io/v1alpha1/namespaces/default/rayclusters/quicktest` when calling the `get_namespaced_custom_object` so I think there is something wrong with this function

**Comment by jbusche** [comment]
I tried it out on my OC 4.12 cluster and it's looked good - but Carson noticed a min/max workers aren't matching what was defined:
cluster.details() confirms 2/2 workers
```
 ╭────────────────────────────────────────────────────────────────╮ 
 │   Name                                                         │ 
 │   mnisttest                                        Active ✅   │ 
 │                                                                │ 
 │   URI: ray://mnisttest-head-svc.default.svc:10001              │ 
 │                                                                │ 
 │   Dashboard🔗                                                  │ 
 │                                                                │ 
 │                      Cluster Resources                         │ 
 │   ╭─ Workers ──╮  ╭───────── Worker specs(each) ─────────╮     │ 
 │   │  Min  Max  │  │  Memory      CPU         GPU         │     │ 
 │   │            │  │                                      │     │ 
 │   │  2    2    │  │  2~2         2           0           │     │ 
 │   │            │  │                                      │     │ 
 │   ╰────────────╯  ╰──────────────────────────────────────╯     │ 
 ╰────────────────────────────────────────────────────────────────╯
```
But the cluster.config is reporting just 1 each:
```
ClusterConfiguration(name='mnisttest', namespace='default', head_info=[], machine_types=[], min_cpus=2, max_cpus=2, min_worker=1, max_worker=1, min_memory=2, max_memory=2, gpu=0, template='/opt/app-root/src/codeflare-sdk/src/codeflare_sdk/templates/base-template.yaml', instascale=False, envs={}, image='ghcr.io/foundation-model-stack/base:ray2.1.0-py38-gpu-pytorch1.12.0cu116-20221213-193103', local_interactive=False)
```

**Review by jbusche — APPROVED** [review]
LGTM
Testing on my cluster, it looked good.  Thanks Carson

**Pull request merged**